scene/resources/resource_format_text.cpp

```diff
@@ -1823,6 +1823,9 @@ void ResourceFormatSaverTextInstance::_find_resources(const Variant &p_variant,
 			List<Variant> keys;
 			d.get_key_list(&keys);
 			for (const Variant &E : keys) {
+				// Of course keys should also be cached, after all we can't prevent users from using resources as keys, right?
+				// See also ResourceFormatSaverBinaryInstance::_find_resources (when p_variant is of type Variant::DICTIONARY)
+				_find_resources(E);
 				Variant v = d[E];
 				_find_resources(v);
```

Comment:
Sorry, didn't read your full explanation, but I doubt this is the right fix. `_find_resources` sounds like an expensive function to call, so we shouldn't call it twice. _If this is not an issue_, I recommend putting another comment that explains _why_ this is called twice.

Reply:
`_find_resources` is not an expensive function, on the contrary, it is necessary. The save will fail without the necessary caching of all sub-resources that need to be cached.